crates/partition-store/src/partition_store_manager.rs

```diff
@@ -192,27 +196,65 @@ impl PartitionStoreManager {
             .await?;
 
         assert!(self.rocksdb.inner().cf_handle(&cf_name).is_some());
-        let partition_store = PartitionStore::new(
+        self.create_partition_store(guard, partition_id, partition_key_range, cf_name)
+            .await
+    }
+
+    /// Creates and registers a new partition store
+    async fn create_partition_store(
+        &self,
+        mut guard: MutexGuard<'_, PartitionStoreLookup>,
+        partition_id: PartitionId,
+        partition_key_range: RangeInclusive<PartitionKey>,
+        cf_name: CfName,
+    ) -> Result<PartitionStore, RocksError> {
+        let mut partition_store = PartitionStore::new(
             self.rocksdb.clone(),
             cf_name,
             partition_id,
             partition_key_range,
         );
-        guard.live.insert(partition_id, partition_store.clone());
+
+        // This method assumes that it is the only path to construct a partition store, and there
+        // are no unflushed writes to the underlying column family on open. This holds as long as
+        // all writes to the underlying CF happen through a partition store tracked in the lookup
+        // map, and one partition store handles all writes to the underlying CF. If this changes,
+        // the persisted LSN determination logic below must be updated appropriately.
+        let live_store = guard.insert(partition_id, partition_store.clone());
+        assert!(
+            live_store.is_none(),
+            "create_partition_store found an open partition"
+        );
+
+        let applied_lsn = match partition_store.get_applied_lsn().await {
+            Ok(Some(applied_lsn)) => applied_lsn,
+            Ok(None) => Lsn::INVALID,
+            Err(err) => {
+                warn!(
+                    %partition_id,
+                    "Failed reading the applied LSN for partition store: {}", err
+                );
+                Lsn::INVALID
+            }
+        };
+        self.persisted_lsns.insert(partition_id, applied_lsn);
 
         Ok(partition_store)
     }
 
+    pub fn get_persisted_lsn(&self, partition_id: PartitionId) -> Option<Lsn> {
```

Comment:
The only user of this for now `PartitionProcessorManager`, which reads the persisted LSNs when constructing `PartitionProcessorStatus`.